harness/determined/_train_context.py

```diff
@@ -196,58 +264,147 @@ class DistributedContext:
 
     def __init__(
         self,
-        env: det.EnvContext,
-        hvd_config: horovod.HorovodContext,
+        rank_info: RankInfo,
```

Comment:
No, because RendezvousInfo is a machine-level data structure that is:
* independent of the launch layer
* consumed by the launch layer

while the RankInfo is a worker-level data structure that is:
* dependent on the launch layer
* consumed by the worker processes